include/openthread/error.h

```diff
@@ -349,6 +349,12 @@ typedef enum otError
      */
     OT_ERROR_LINK_MARGIN_LOW = 34,
 
+    /**
+     * Message was evicted.
+     *
+     */
+    OT_ERROR_EVICTED = 35,
```

Comment:
What is the difference between `OT_ERROR_EVICTED` and `OT_ERROR_DROP`?

Reply:
`EVICT` is more specific drop indicating that message was explicitly removed to make room for a higher priority message. `DROP` can mean different things: For example, when device is detached, we drop a direct message from  `MeshForwarder::UpdateIp6Route()`) (unless it's to a link-local unicast or multicast) , or if we cannot reach dest from `CheckReachability()`, etc.

I can add it in the document about `EVIT`. What do you think?

Reply:
What about the case where there is a newer message that supersedes an older message (e.g. with MLE messages)? Should that be classified as `OT_ERROR_EVICTED` or `OT_ERROR_DROP`?

Reply:
I'd suggest we go for `DROP` for such cases, main reason being we want `EVICT` to indicate to user that device ran out of memory/buffer (and thus the need for eviction).